app/src/org/commcare/activities/FormEntryActivity.java

```diff
@@ -442,36 +442,42 @@ private boolean processIntentResponse(Intent response, boolean wasIntentCancelle
         return wasAnswerSet;
     }
 
-
     @Override
-    public boolean onPrepareOptionsMenu(Menu menu) {
```

Comment:
Heads up that the reason this is in "onPrepare" is so that it gets rebuilt when langauges change. If it's moved to onCreate() those triggers no longer fire each time, so the langauges won't update

Reply:
Makes sense. Added a call to `invalidateOptionsMenu` on language change so that `onCreateOptionsMenu` gets called again.